File: github-mvi-android/cache/src/main/java/com/rightpoint/mvi/example/cache/room/DatabaseModule.java
package com.rightpoint.mvi.example.cache.room;

import android.app.Application;

import javax.inject.Singleton;

import androidx.room.Room;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class DatabaseModule {
    @Provides
    @Singleton
    static AppDatabase providesDatabase(Application app) {
        return Room.databaseBuilder(app, AppDatabase.class, "github-mvi-android.db").build();
    }
}

File: github-mvi-android/cache-room/src/main/java/com/rightpoint/github/mvi/cache/room/DatabaseModule.kt
package com.rightpoint.github.mvi.cache.room

import android.app.Application
import android.arch.persistence.room.Room
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
object DatabaseModule {
    @Provides
    @Singleton
    @JvmStatic
    fun providesDatabase(app: Application): AppDatabase {
        return Room.databaseBuilder(app, AppDatabase::class.java, "github-mvi-android-db")
            .fallbackToDestructiveMigration()
            .build()
    }
}
